test/components/Homepage_test.js

```diff
@@ -240,23 +243,58 @@ describe("HomePage component", () => {
           ],
           settings: {},
           session: { authenticated: false, serverInfo: DEFAULT_SERVERINFO },
-        });
-
-        expect(node.querySelector("#root_server").value).eql(
+        };
+        const wrapper = mount(<HomePage {...props} />);
+        expect(wrapper.find("input#root_server").prop("value")).eql(
           "http://server.test/v1"
         );
-        expect(node.querySelector("#root_authType").value).eql("basicauth");
+        expect(wrapper.find("input#root_authType").prop("value")).eql(
+          "basicauth"
+        );
 
         // Changing the server to another element from the history.
-        Simulate.change(node.querySelector("#root_server"), {
+        wrapper.find("input#root_server").simulate("change", {
           target: { value: "http://test.server/v1" },
         });
-        expect(node.querySelector("#root_server").value).eql(
+        expect(wrapper.find("input#root_server").prop("value")).eql(
           "http://test.server/v1"
         );
         // authType is reset to "anonymous" while we wait for the server info
         // (capabilities)
-        expect(node.querySelector("#root_authType").value).eql("anonymous");
+        expect(wrapper.find("input#root_authType").prop("value")).eql(
+          "anonymous"
+        );
+
+        // Simulate a `getServerInfo` response that updated the `capabilities`.
+        const prevProps = wrapper.props();
+        wrapper.setProps({
+          ...prevProps,
+          session: {
+            ...prevProps.session,
+            serverInfo: {
+              ...prevProps.session.serverInfo,
+              capabilities: {
+                ...prevProps.session.serverInfo.capabilities,
+                basicauth: "some basic auth info",
+                ldap: "some ldap auth info",
+                fxa: "some fxa auth info",
+                openid: {
+                  providers: [
+                    {
+                      name: "google",
+                    },
+                  ],
+                },
+              },
+            },
+          },
+        });
```

Comment:
This might be a dumb question, but is setting a giant blob of props the "done thing" in Enzyme? I would have expected some kind of simulated action or something.

Reply:
So what Enzyme gives us is the ability to even do `setState` at all (thus changing the state during the test).

Another way then manually setting the props could have been to
1/ use the `HomePage` **container** (and not component)
2/ connect it to a (fake) store
3/ dispatch the action on the store so that it changes the state of the `HomePage` container, and trickles down as props in the `AuthForm`

I feel this would couple code a lot more, and maybe add some indirection (the test wouldn't be as "unit-y"). The compromise is to have the test coupled to the "props/state structure" which is also a problem. Not sure what's the best, what do you think?

Reply:
I don't have a strong opinion, this is fine.